### PR TITLE
Histogram refactor and mindepth

### PR DESCRIPTION
**Pull request by HGinnerup**

### CODE REVIEW

File: Jantimizer/Histograms/ValueCount.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Histograms
```
Comment (kris701) [inline Jantimizer/Histograms/ValueCount.cs:7]: Should this maybe not be in the `Models` namespace?

**Review by kris701 — CHANGES_REQUESTED** [review]

File: Jantimizer/Histograms/Managers/PostgresEquiDepthHistogramManager.cs [hunk 2]
```diff
@@ -85,11 +86,13 @@ private async Task<DataTable> GetAttributenamesForTable(string tableName)
         private async Task AddHistogramForAttribute(DataRow row, string tableName)
         {
             string attributeName = $"{row["column_name"]}".ToLower();
-            var attributeValues = await DbConnector.CallQuery($"SELECT {attributeName} FROM {tableName}");
-            if (attributeValues.Tables.Count > 0)
+            DataSet sortedGroupsDs = await DbConnector.CallQuery($"SELECT {attributeName} AS val, count({attributeName}) FROM {tableName} WHERE {attributeName} IS NOT NULL GROUP BY {attributeName} ORDER BY {attributeName} ASC");
```
[the remark this answers is not in the document]
Comment (HGinnerup) [reply Jantimizer/Histograms/Managers/PostgresEquiDepthHistogramManager.cs:89]: > Also, is there any particular reason why you are using `ORDER BY`? I thought the order didnt really matter here.

It does matter eventually, so I order it on the database to skip it in c#

**Review by jamadaha — CHANGES_REQUESTED** [review]

File: Jantimizer/Histograms/Managers/PostgresEquiDepthHistogramManager.cs [hunk 3]
```diff
@@ -85,15 +86,32 @@ private async Task<DataTable> GetAttributenamesForTable(string tableName)
...
+                GROUP BY {attributeName}
+                ORDER BY {attributeName} ASC
+            ");
+
+            List<ValueCount> sortedGroups = GetValueCounts(sortedGroupsDs, "val", "COUNT").ToList();
+
+            IHistogram newHistogram = new HistogramEquiDepth(tableName, attributeName, Depth);
+            newHistogram.GenerateHistogram(sortedGroups);
+            Histograms.Add(newHistogram);
         }
```
Comment (kris701) [inline Jantimizer/Histograms/Managers/PostgresEquiDepthHistogramManager.cs:104]: Why was this only added to the Postgres manager and not the MySQL manager?

**Review by kris701 — CHANGES_REQUESTED** [review]

File: Jantimizer/Histograms/Managers/PostgresEquiDepthHistogramManager.cs (hunk 3 above)
Comment (HGinnerup) [reply Jantimizer/Histograms/Managers/PostgresEquiDepthHistogramManager.cs]: I forgot it at the time, but it already works with MySQL continuing in its own way,
so I vote for letting that be part of #108

**Review by jamadaha — APPROVED** [review]

**Review by kris701 — APPROVED** [review]

**Pull request merged**